prow/config/inrepoconfig.go

```diff
@@ -147,6 +148,11 @@ func prowYAMLGetter(
 	// change was a fast-forward merge. So we need to dedupe it with sets.
 	repoOpts.NeededCommits = sets.New(baseSHA)
 	repoOpts.NeededCommits.Insert(headSHAs...)
+	branchesToRetarget := make(map[string]string)
+	if len(baseBranch) > 0 {
+		branchesToRetarget[baseBranch] = baseSHA
+	}
+	repoOpts.BranchesToRetarget = branchesToRetarget
```

Comment:
nit: Avoid initializing a map if we don't need one.  This can be simplified to the following:
```suggestion
	if baseBranch != "" {
		repoOpts.BranchesToRetarget = map[string]string{baseBranch: baseSHA}
	}
```

I'm curious why a map is needed/preferred rather than just storing the `baseRef` and `baseSHA` fields in `repoOpts`.

Reply:
> I'm curious why a map is needed/preferred rather than just storing the baseRef and baseSHA fields in repoOpts.

Two reasons:

1. A map, which has a clear "key to value" relationship was more readable than having separate explicit fields. The `baseRef` field is useless without the `baseSHA`, but if we have separate fields for them it is possible that we could construct a buggy `repoOpts` that has a `baseRef` without a `baseSHA`. This is less likely to happen with a map where the keys are required.
2. I couldn't rule out that we would never want to expand this to cover more than just the base branch in the future.

WRT avoiding the map initialization, SGTM. Will update.